test-suite/coq-makefile/findlib-package-unpacked/run.sh

```diff
@@ -16,5 +16,7 @@ coq_makefile -f _CoqProject -o Makefile
 cat Makefile.conf
 cat Makefile.local
 make -C findlib/foo
-make
+if which ocamlopt >/dev/null 2>&1; then
+    make
+fi
```

Comment:
Shouldn't coq_makefile do the right thing here?

Reply:
I think so.

@treinen did you see this test fail on byte only architectures, didn't you? Would you mind posting the error?

Reply:
I am not so sure, but in fact I do not know what the purpose of the generated makefile is.

In general, I would expect a Makefile to do the right thing, independent of the architecture. In case of building an executable that would mean building a native code or a bc executable, depending on the architecture. In case of compiling a module, however, I would find it strange that a make gives me in one case a cmx and in other cases a cmo.

This two expectations are of course somewhat contradictory.

The error messages are
````
"/usr/bin/ocamlfind" opt -c foolib.ml
ocamlfind: Not supported in your configuration: ocamlopt
make[5]: *** [Makefile:7: all] Error 2
make[5]: Leaving directory '/<<PKGBUILDDIR>>/test-suite/coq-makefile/findlib-package-unpacked/_test/findlib/foo'
````
and
````
"/usr/bin/ocamlfind" opt      -c  -thread -rectypes -w +a-4-9-27-41-42-44-45-48-58   -safe-string -strict-sequence -I src  -I "/<<PKGBUILDDIR>>//config"  -I "/<<PKGBUILDDIR>>//lib"  -I "/<<PKGBUILDDIR>>//clib"  -I "/<<PKGBUILDDIR>>//kernel"  -I "/<<PKGBUILDDIR>>//library"  -I "/<<PKGBUILDDIR>>//engine"  -I "/<<PKGBUILDDIR>>//pretyping"  -I "/<<PKGBUILDDIR>>//interp"  -I "/<<PKGBUILDDIR>>//gramlib"  -I "/<<PKGBUILDDIR>>//gramlib/.pack"  -I "/<<PKGBUILDDIR>>//parsing"  -I "/<<PKGBUILDDIR>>//proofs"  -I "/<<PKGBUILDDIR>>//tactics"  -I "/<<PKGBUILDDIR>>//toplevel"  -I "/<<PKGBUILDDIR>>//printing"  -I "/<<PKGBUILDDIR>>//ide"  -I "/<<PKGBUILDDIR>>//stm"  -I "/<<PKGBUILDDIR>>//vernac"  -I "/<<PKGBUILDDIR>>//plugins/btauto"  -I "/<<PKGBUILDDIR>>//plugins/cc"  -I "/<<PKGBUILDDIR>>//plugins/derive"  -I "/<<PKGBUILDDIR>>//plugins/extraction"  -I "/<<PKGBUILDDIR>>//plugins/firstorder"  -I "/<<PKGBUILDDIR>>//plugins/fourier"  -I "/<<PKGBUILDDIR>>//plugins/funind"  -I "/<<PKGBUILDDIR>>//plugins/ltac"  -I "/<<PKGBUILDDIR>>//plugins/micromega"  -I "/<<PKGBUILDDIR>>//plugins/nsatz"  -I "/<<PKGBUILDDIR>>//plugins/omega"  -I "/<<PKGBUILDDIR>>//plugins/rtauto"  -I "/<<PKGBUILDDIR>>//plugins/setoid_ring"  -I "/<<PKGBUILDDIR>>//plugins/ssr"  -I "/<<PKGBUILDDIR>>//plugins/ssrmatching"  -I "/<<PKGBUILDDIR>>//plugins/syntax" -warn-error +a-3   src/file1.ml
ocamlfind: Not supported in your configuration: ocamlopt
make[5]: *** [Makefile:626: src/file1.cmx] Error 2
make[5]: Leaving directory '/<<PKGBUILDDIR>>/test-suite/coq-makefile/camldep/_test'
````

See https://buildd.debian.org/status/fetch.php?pkg=coq&arch=armel&ver=8.11.0-1&stamp=1583482150&raw=0 for the details.

Reply:
Here is the list of the architectures where I have observed this kind of failures. The names of the archictures are the ones used by debian:

armel mipsel hppa ia64 powerpc riscv64 x32

These are all bytecode architectures.